components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -755,4 +762,110 @@ public boolean validateAxis2ClientRepo(String input) throws AxisFault {
+                messageContext = messageConsumer.receive();
+                msg = messageContext.getEnvelope().toString();
+            } catch (SynapseException e) {
+                log.error("Failed to get message : ", e);
+            }
+        }
+        return  msg;
+    }
+
+    /**
+     * Pops the message from the associated queue
+     *
+     * @param processorName Message Processor Name
+     */
+    public void popMessage(String processorName) {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+
+        popMessageFromQueue(messageConsumer);
+        messageConsumer.cleanup();
+    }
+
+    private  void popMessageFromQueue(MessageConsumer consumer) throws SynapseException{
+        MessageConsumer messageConsumer = consumer;
+
+        try {
+            messageConsumer.receive();
+            messageConsumer.ack();
+        } catch (SynapseException e) {
+            log.error("Failed to pop message : ",e);
+        }
+    }
+
+    /**
+     * Pops the message and enqueues it to a specified queue
+     *
+     * @param processorName Message Processor Name
+     * @param storeName Name of the store to enqueue the message
+     */
+
+    public void popAndEnqueueMessage(String processorName, String storeName){
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+        MessageProducer messageProducer = configuration.getMessageStore(storeName).getProducer();
+
+        popAndEnqueueMessageToStore(messageProducer, messageConsumer);
+        messageConsumer.cleanup();
+    }
+
+    private void popAndEnqueueMessageToStore(MessageProducer producer, MessageConsumer consumer) throws SynapseException
+    {
+        MessageProducer messageProducer = producer;
+        MessageConsumer messageConsumer = consumer;
+        MessageContext messageContext;
+
+        try {
+            messageContext = messageConsumer.receive();
+            messageProducer.storeMessage(messageContext);
+            messageConsumer.ack();
+        } catch (SynapseException e) {
+            log.error("Failed to enqueue the message : ",e);
+        }
+
+    }
+
+    /**
+     * Gets the MessageConsumer associated with the specified processor
+     *
+     * @param configuration SynapseConfiguration
+     * @param processorName Message Processor Name
+     * @return <code>messageConsumer</code> object associated with specified processor
+     */
+
+    private MessageConsumer getMessageConsumer(SynapseConfiguration configuration, String processorName) {
+        MessageProcessor processor = configuration.getMessageProcessors().get(processorName);
+        final String messageStoreName = processor.getMessageStoreName();
+        MessageConsumer messageConsumer = configuration.getMessageStore(messageStoreName).getConsumer();
```

Comment:
`messageConsumer` variable seems to be redundant here. Shall we return inline?
`        return configuration.getMessageStore(messageStoreName).getConsumer();`